### DISCUSSION TITLE
Error in installation

### DISCUSSION THREAD

**Discussion by crosassalazar** [FrederickHuangLin/ANCOMBC#71] [Q&A]
Hello:
I tried installing ANCOM-BC but encountered an error. I'm running R 4.2.1 and Bioconductor 3.15. I'd appreciate any ideas/suggestions on what the issue is. Thanks for developing this package!

> BiocManager::install("ANCOMBC", force=TRUE)
'getOption("repos")' replaces Bioconductor standard repositories, see '?repositories' for
details

replacement repositories:
    CRAN: https://cran.rstudio.com/

Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)
Installing package(s) 'ANCOMBC'

  There is a binary version available but the source version is later:
        binary source needs_compilation
ANCOMBC  1.6.1  1.6.2             FALSE

installing the source package ‘ANCOMBC’

trying URL 'https://bioconductor.org/packages/3.15/bioc/src/contrib/ANCOMBC_1.6.2.tar.gz'
Content type 'application/x-gzip' length 928299 bytes (906 KB)
==================================================
downloaded 906 KB

* installing *source* package ‘ANCOMBC’ ...
** using staged installation
** R
** inst
** byte-compile and prepare package for lazy loading
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘gld’
Calls: <Anonymous> ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
ERROR: lazy loading failed for package ‘ANCOMBC’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/ANCOMBC’

The downloaded source packages are in
	‘/private/var/folders/mr/k4_62d5n7tndn56mtt1692940000gn/T/Rtmp3MHnE8/downloaded_packages’
Warning message:
In install.packages(...) :
  installation of package ‘ANCOMBC’ had non-zero exit status
> library(ANCOMBC)
Error in library(ANCOMBC) : there is no package called ‘ANCOMBC’

**Comment by FrederickHuangLin** [comment]
Hi @christianrs5,

Could you try `BiocManager::install("ANCOMBC")` (without `force = TRUE`) and see if the issue persists? 

I tried on both PC and MacBook, it seems to work.

Best,
Huang

**Reply by crosassalazar** [reply]
It didn't work, but only gave me one error that the gld package couldn't be installed. I installed it separately (prior to ANCOMBC) and now it works. Looking forward to using your package now. Thanks!
